src/core-layers/grid-layer/grid-layer.js

```diff
@@ -56,114 +69,238 @@ export default class GridLayer extends CompositeLayer {
   initializeState() {
     this.state = {
       layerData: [],
-      sortedBins: null,
-      valueDomain: null
+      sortedColorBins: null,
+      sortedElevationBins: null,
+      colorValueDomain: null,
+      elevationValueDomain: null,
+      colorScaleFunc: nop,
+      elevationScaleFunc: nop,
+      dimensionUpdaters: this.getDimensionUpdaters()
     };
   }
 
   updateState({oldProps, props, changeFlags}) {
+    const dimensionChanges = this.getDimensionChanges(oldProps, props);
+
     if (changeFlags.dataChanged || this.needsReProjectPoints(oldProps, props)) {
       // project data into hexagons, and get sortedBins
       this.getLayerData();
-      this.getSortedBins();
+    } else if (dimensionChanges) {
+      dimensionChanges.forEach(f => typeof f === 'function' && f.apply(this));
+    }
+  }
 
-      // this needs sortedBins to be set
-      this.getValueDomain();
-    } else if (this.needsReSortBins(oldProps, props)) {
+  needsReProjectPoints(oldProps, props) {
+    return oldProps.cellSize !== props.cellSize || oldProps.getPosition !== props.getPosition;
```

Comment:
This line is probably my only big concern with this PR. This goes against how deck.gl layers usually work. Supplying a new accessor should not trigger an update. We provide fine grained control over that with `updateTriggers`.

I have written a new article about [updates](https://github.com/uber/deck.gl/blob/master/docs/advanced/updates.md) for the upcoming release to make this more clear to users.

Reply:
will remove this and rely on user supply updateTriggers